// tsdoctest/OtherModule.ts
/**
 * This is a documentation for the OtherModule module. It depends on the MyModule module
 * and contains the [[OtherClass]] class.
 */

 /** */
import * as myMd from 'MyModule';
import { ExampleModule1Class } from 'example-module1';
import { ExampleModule2Class } from 'example-module2';

/**
 * This class is inherited from the [[DerivedClass]] class.
 */
export class OtherClass extends myMd.DerivedClass {
    /**
     * This is an alternative to the [[DerivedClass.theMethod]] method and
     * the [[mainProp]] property.
     * @param value A [[DerivedClass]] instance to {@link DerivedClass} process.
     * @returns The resulting [[MyClass]] instance.
     */
    theOtherMethod(value: myMd.DerivedClass): myMd.MyClass {
        // console.log(abc);
        return value;
    }

}

/**
 * This class is inherited from the [[ExampleModule1Class]] class.
 */
export class OtherClass2 extends ExampleModule1Class{
    /**
     * This is an alternative to the [[DerivedClass.theMethod]] method and
     * the [[mainProp]] property.
     */
    theSampleMethod(value: ExampleModule1Class): ExampleModule1Class {
        
        return value;
    }

}

/**
 * This class is inherited from the [[ExampleModule2Class]] class.
 */
export class OtherClass3 extends ExampleModule2Class{
    /**
     * This is an alternative to the [[DerivedClass.theMethod]] method and
     * the [[mainProp]] property.
     */
    theSampleMethod(value: ExampleModule2Class): ExampleModule2Class {
        
        return value;
    }

}
